admin/tests/playwright/admin.spec.ts

```diff
@@ -1,14 +1,41 @@
-import { expect, test } from '@playwright/test';
+import { expect, test as base } from '@playwright/test';
+import * as fs from 'fs';
+import * as path from 'path';
 
-test.beforeEach(async ({ page }) => {
+export const test = base.extend({
+	context: async ({ context }, use) => {
+		const sessionStorage = JSON.parse(
+			fs.readFileSync(path.join(__dirname, 'sessionStorage.json'), { encoding: 'utf8' })
+		);
+		await context.addInitScript((storage: string) => {
+			for (const [key, value] of Object.entries(storage)) {
+				window.sessionStorage.setItem(key, value as string);
+			}
+		}, sessionStorage);
+		await use(context);
+	},
+});
+
+test.beforeAll(async ({ browser }) => {
+	console.log('Logging in and writing sessionStorage.json file.');
```

Comment:
@mkue: After your change `auth.setPersistence(browserSessionPersistence);`.

Would `await page.context().storageState({ path: 'storageState.json' });` also work to save it? https://playwright.dev/docs/auth#reuse-signed-in-state

And something like
```
test.describe(() => {
  test.use({ storageState: 'userStorageState.json' });

  test('user test', async ({ page }) => {
    // page is signed in as a user.
  });
});
```
to use it?

Reply:
Unfortunately not. Playwright uses session storage (https://playwright.dev/docs/auth#session-storage) and not regular cookies (where we could use storage state).

Reply:
Ahhh I see 👌